core/value.mli

```diff
@@ -55,16 +55,20 @@ and xml = xmlitem list
 type table = (database * string) * string * string list list * Types.row
   [@@deriving show]
 
+type primitive_value_basis =  [
+  | `Bool of bool
+  | `Char of char
+  | `Float of float
+  | `Int of int
+  | `XML of xmlitem
+  | `String of string ]
+  [@@deriving show]
+
 type primitive_value = [
-| `Bool of bool
-| `Char of char
-| `Database of (database * string)
-| `Table of table
-| `Float of float
-| `Int of int
-| `XML of xmlitem
-| `String of string ]
-[@@deriving show]
+  | primitive_value_basis
+  | `Database of (database * string)
+  | `Table of table ]
+  [@@deriving show]
```

Comment:
I don't like this. Can we keep it as a single data type? When we proceed with removing polymorphic invariants - and I hope this will happen at some point - this refactoring will only complicate things.

Reply:
This is introduced by #769.